compiler/passes/insertWideReferences.cpp

```diff
@@ -1482,13 +1484,23 @@ static void narrowWideClassesThroughCalls()
             // Insert a local check because we cannot reflect any changes
             // made to the class back to another locale
             if (!fNoLocalChecks)
-              stmt->insertBefore(new CallExpr(PRIM_LOCAL_CHECK, sym->copy()));
+              stmt->insertBefore(new CallExpr(PRIM_LOCAL_CHECK, sym->copy(), buildCStringLiteral("cannot access remote data in local block")));
 
             // If we pass an extern class to an extern/export function,
             // we must treat it like a reference (this is by definition)
             stmt->insertBefore(new CallExpr(PRIM_MOVE, var, sym->copy()));
           }
           else if (narrowType.isRef() || narrowType.type()->symbol->hasFlag(FLAG_DATA_CLASS)) {
+
+            // Insert a local check because we cannot pass narrow references to
+            // remote data to external routines
+            if (!fNoLocalChecks) {
+              if (fn->hasFlag(FLAG_EXTERN))
+                stmt->insertBefore(new CallExpr(PRIM_LOCAL_CHECK, sym->copy(), buildCStringLiteral(astr("references to remote data cannot be passed to external routines like '", fn->name, "'"))));
+              else if (fn->hasFlag(FLAG_EXPORT))
+                stmt->insertBefore(new CallExpr(PRIM_LOCAL_CHECK, sym->copy(), buildCStringLiteral(astr("references to remote data cannot currently be passed to exported routines like '", fn->name, "' (though this is a bug)"))));
```

Comment:
I'd definitely leave the "Though this is a bug" until we're all convinced of that. "Cannot currently" is sufficient I think.

Reply:
You could say "(Though this may change as the implementation improves)" or something along those lines.

Reply:
With fresh eyes today, I agree.